constraints/src/main/java/gov/nasa/jpl/aerie/constraints/time/Windows.java

```diff
@@ -1,267 +1,334 @@
 package gov.nasa.jpl.aerie.constraints.time;
 
 import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+import org.apache.commons.lang3.tuple.Pair;
 
-import java.util.Iterator;
+import java.util.Arrays;
 import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
-import java.util.stream.Collectors;
+import java.util.Spliterator;
+import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
-import static gov.nasa.jpl.aerie.constraints.time.Window.Inclusivity.Inclusive;
+public final class Windows extends IntervalMap<Boolean> {
 
-public final class Windows implements Iterable<Window> {
-  private final IntervalSet<WindowAlgebra, Window> windows = new IntervalSet<>(new WindowAlgebra());
-
-  public Windows() {}
-
-  public Windows(final Windows other) {
-    this.windows.addAll(other.windows);
+  public Windows() {
+    super();
   }
 
-  public Windows(final List<Window> windows) {
-    for (final var window : windows) this.add(window);
+  public Windows(IntervalAlgebra a) { //only for testing.
+    super(a);
   }
 
-  public Windows(final Window... windows) {
-    for (final var window : windows) this.add(window);
-  }
-
-  public Spans intoSpans() {
-    return new Spans(this);
+  public Windows(final Windows other) {
+    super(other);
   }
 
-  public void add(final Window window) {
-    this.windows.add(window);
+  @SafeVarargs
+  public Windows(final Pair<Interval, Boolean>... windows) {
+    super(windows);
   }
 
-  public void addAll(final Windows other) {
-    this.windows.addAll(other.windows);
+  public Windows(final Interval interval, final boolean value) {
+    super(interval, value);
   }
 
-  public void addPoint(final long quantity, final Duration unit) {
-    this.add(Window.at(quantity, unit));
+  private Windows(final IntervalMap<Boolean> from) {
+    super(from.alg, from.segments);
   }
 
-  public static Windows union(final Windows left, final Windows right) {
-    final var result = new Windows(left);
-    result.addAll(right);
+  public static Windows definedEverywhere(final Interval interval, final boolean value) {
```

Comment:
I don't like it either. Do you have a better name?

Reply:
Not yet :/ I spent a little time trying to see if `definedEverywhere` could simply be removed outright (which would obviate the need for any name at all), since it seems a little semantically questionable to me. But I need to better understand the places it's used.

Reply:
Given my new feelings about the validity and intent for `definedEverywhere`, I'd advocate for replacing it with a chained pair of methods `constant(!value).set(I, value)`. It's short enough, and self-explanatory enough, that we can inline it anywhere the current `definedEverywhere` is used, and it's built out of parts that apply to any `IntervalMap`, not just boolean ones.

Reply:
Ok I like that too